velox/functions/sparksql/tests/ArrayShuffleTest.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/VectorReaders.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+using namespace facebook::velox::test;
+
+class ArrayShuffleTest : public SparkFunctionBaseTest {
+ protected:
+  template <typename T>
+  VectorPtr
+  testShuffle(const VectorPtr& input, int64_t seed, int32_t partitionId = 0) {
+    setSparkPartitionId(partitionId);
+    return evaluate(
+        fmt::format("shuffle(c0, {})", seed), makeRowVector({input}));
+  }
+
+  template <typename T>
+  void assertDifferentSequence(
+      const VectorPtr& result,
+      const VectorPtr& expected) {
+    DecodedVector decodedResult(*result.get());
+    exec::VectorReader<Array<T>> readerResult(&decodedResult);
+    DecodedVector decodedExpected(*expected.get());
+    exec::VectorReader<Array<T>> readerExpected(&decodedExpected);
+    for (auto i = 0; i < decodedExpected.size(); i++) {
+      auto resultArray = readerResult[i].materialize();
+      auto expectedArray = readerExpected[i].materialize();
+      if (resultArray.size() > 2) {
+        ASSERT_FALSE(std::equal(
+            resultArray.begin(), resultArray.end(), expectedArray.begin()));
+      }
+      ASSERT_TRUE(std::is_permutation(
+          resultArray.begin(), resultArray.end(), expectedArray.begin()));
+    }
+  }
+};
+
+TEST_F(ArrayShuffleTest, basic) {
+  auto input = makeArrayVector<int64_t>({{1, 2, 3, 4, 5}});
+  auto result = makeArrayVector<int64_t>({{3, 5, 4, 1, 2}});
+  auto stringInput = makeArrayVector<std::string>({{"a", "b", "c", "d"}});
+  auto stringResult = makeArrayVector<std::string>({{"a", "c", "b", "d"}});
+  assertEqualVectors(testShuffle<int64_t>(input, 0), result);
+  assertEqualVectors(testShuffle<std::string>(stringInput, 0), stringResult);
+
+  // Assert results are different with different seeds / partition ids.
+  assertDifferentSequence<int64_t>(
```

Comment:
Since this function is deterministic we can simply hard-code expected results. This would make the test easier to read and also make it stronger. The test would fail if function is modified in a way that makes it non-deterministic.